python/lolopy/tests/test_learners.py

```diff
@@ -181,6 +181,51 @@ def test_adjust_rtree_learners(self):
         rf.fit(X[:16, :], y[:16])
         self.assertLess(r2_score(y, rf.predict(X)), 1.0)  # Should not fit the whole dataset perfectly
 
+class TestExtraRandomTrees(TestCase):
+
+    def test_extra_random_trees_regressor(self):
+        rf = ExtraRandomTreesRegressor()
+
+        # Train the model
+        X, y = load_boston(True)
+
+        # Make sure we get a NotFittedError
+        with self.assertRaises(NotFittedError):
+            rf.predict(X)
+
+        # Fit the model
+        rf.fit(X, y)
+
+        # Run some predictions
+        y_pred = rf.predict(X)
+        self.assertEqual(len(y_pred), len(y))
+
+        # Test the ability to get importance scores
+        y_import = rf.get_importance_scores(X[:100, :])
+        self.assertEqual((100, len(X)), y_import.shape)
+
+        # Basic test for functionality. R^2 above 0.98 was measured on 27Dec18
+        score = r2_score(y_pred, y)
+        print('R^2:', score)
+        self.assertGreater(score, 0.98)
+
+        # Test with weights (make sure it doesn't crash)
+        rf.fit(X, y, [1.0]*len(y))
+
+        # Make sure feature importances are stored
+        self.assertEqual(np.shape(rf.feature_importances_), (X.shape[1],))
+        self.assertAlmostEqual(1.0, np.sum(rf.feature_importances_))

```

Comment:
Despite having a similar name, `get_importance_scores` is intended to capture the impact of training data, which is related but not identical to the gain scores we use to communicate feature importance. I would like to tidy up this terminology, and document it more transparently, but that's not of high relative priority and should be addressed in a separate PR.

Reply:
Ah, thanks for clarifying. I think the existence of this conversation indicates that more expressive naming is indeed desirable. If not in this PR, could you file a corresponding issue so that this does not get lost?

Reply:
https://github.com/CitrineInformatics/lolo/issues/224